docs/building.md

```diff
@@ -0,0 +1,125 @@
+1. Create `build/Dockerfile.<new-component>`
+    > A simple way to do this is to copy `build/Dockerfile.agent` and replace `agent` with whatever `<new-component>` is.
+1. Create `.github/workflows/build-<new-component>-container.yml`
+    > A simple way to do this is to copy `.github/workflows/build-agent-container.yml` and replace `agent` with whatever `<new-component>` is.
+
+
+## Intermediate builds
+There are the intermediate containers:
+* [rust-crossbuild](https://github.com/orgs/deislabs/packages/container/package/akri%2Frust-crossbuild)
+* [opencvsharp-build](https://github.com/orgs/deislabs/packages/container/package/akri%2Fopencvsharp-build)
+
+### rust-crossbuild
+This container is used by the Akri cross-build process.  The main purpose of these containers is to provide `Cargo cross` with a Rust build environment that has all the required dependencies installed.  This container can be built locally for all platforms using this command:
+```bash
+BUILD_AMD64=1 BUILD_ARM32=1 BUILD_AMD64=1 make rust-crossbuild
+```
+
+If a change needs to be made to this container, 2 pull requests are needed.
+1. Create PR with desired `rust-crossbuild` changes (new dependencies, etc) AND update `BUILD_RUST_CROSSBUILD_VERSION` in `build/intermediate-containers.mk`.  This PR is intended to create the new version of `rust-crossbuild` (not to use it).
+1. After 1st PR is merged and the new version of `rust-crossbuild` is pushed to ghcr.io/akri, create PR with any changes that will leverage the new version of `rust-crossbuild` AND update `Cross.toml` (the `BUILD_RUST_CROSSBUILD_VERSION` value specified in step 1 should be each label's suffix).  This PR is intended to **use** the new version of `rust-crossbuild`.
+
+### opencvsharp-build
+This container is used by the [onvif-video-broker](https://github.com/orgs/deislabs/packages/container/package/akri%2Fonvif-video-broker) as part of its build process.  The main purpose of this container is to prevent each build from needing to build the OpenCV C# platform.  This container can be built locally for all platforms using this command:
+```bash
+BUILD_AMD64=1 BUILD_ARM32=1 BUILD_AMD64=1 make opencv-base
+```
+
+If a change needs to be made to this container, 2 pull requests are needed.
+1. Create PR with desired `opencvsharp-build` changes (new dependencies, etc) AND update `BUILD_OPENCV_BASE_VERSION` in `build/intermediate-containers.mk`.  This PR is intended to create the new version of `opencvsharp-build` (not to use it).
+1. After 1st PR is merged and the new version of `opencvsharp-build` is pushed to ghcr.io/akri, create PR with any changes that will leverage the new version of `opencvsharp-build` AND update `USE_OPENCV_BASE_VERSION` in `build/akri-containers.mk`.  This PR is intended to **use** the new version of `opencvsharp-build`.
+
+
+## Automated builds usage
+The automated CI builds essentially run these commands, where `<comopnent>` is one of () and `<platform>` is one of (amd64|arm32|arm64):
```

Comment:
is one of () <--- is that intentionally an empty set?

Reply:
oops

Reply:
should be fixed now.